src/buffer.rs

```diff
@@ -243,6 +248,19 @@ impl Buffer {
         Ok(())
     }
 
+    /// Takes other buffer, measures its length and prints this buffer from the point where
+    /// the other stopped.
+    /// Used to implement autosuggestions.
+    pub fn print_rest<W>(&self, out: &mut W, other: &Buffer) -> io::Result<usize>
+        where W: Write
+    {
+        let matching_part = other.data.len();
+        let string: String = self.data.iter().skip(matching_part).cloned().collect();
+        out.write(string.as_bytes())?;
```

Comment:
Should be able to write this as the following in order to eliminate the heap allocations:

```
let data = self.data.as_bytes();
let data = data[matching_part..];
out.write(data)?;
```